.github/workflows/docker.yml

```diff
@@ -0,0 +1,44 @@
+name: Build Docker images
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/docker.yml'
+      - 'docker/**'
+  schedule:
+    - cron: '42 7 * * *' # run at 7:42 UTC (morning) every day
+jobs:
+  build:
+    name: Build and push
+    runs-on: ubuntu-latest
+    env:
+      RUST_IMAGE_TAG: latest
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
```

Comment:
There's an official "docker build and push action" https://github.com/marketplace/actions/build-and-push-docker-images
And also buildah actions, for the connoisseurs: https://github.com/marketplace/actions/buildah-build and https://github.com/redhat-actions/push-to-registry

Reply:
Not sure whether you suggest something particular, but I'd be curious to see an alternative PR implementing the required logic using one of these pre-defined actions - "patches welcome" I guess. :) The tricky part is not to build when there's no need to (unless we want to waste GitHub's resources).

Reply:
Maybe I just didn't understand why creating this on schedule to follow the latest Rust exactly in the containerization job.